src/components/ClientDialog/index.tsx

```diff
@@ -0,0 +1,84 @@
+import {Dialog} from '@lykkex/react-components';
+import classnames from 'classnames';
+import {observable} from 'mobx';
+import {observer} from 'mobx-react';
+import React from 'react';
+import {
+  DialogActionModel,
+  DialogActionType,
+  DialogModel
+} from '../../models/dialogModel';
+
+import './style.css';
+
+export interface ClientDialogProps {
+  dialog: DialogModel;
+  onDialogConfirm: (dialog: DialogModel) => void;
+  onDialogCancel?: (dialog: DialogModel) => void;
+}
+
+export class ClientDialog extends React.Component<ClientDialogProps> {
+  @observable private isConfirmed = this.props.dialog.isConfirmed;
+
+  render() {
+    return (
+      <Dialog
+        visible={this.props.dialog.visible}
+        title={this.props.dialog.header}
+        className={classnames('client-dialog', {
+          'client-dialog_confirm-disabled': !this.isConfirmed
+        })}
+        // tslint:disable-next-line:jsx-no-lambda
+        onCancel={() =>
+          this.props.onDialogCancel &&
+          this.props.onDialogCancel(this.props.dialog)}
+        actions={this.props.dialog.actions
```

Comment:
Move calculation outside component (calculate before return statement) and use value from variable